feature/experimental/gribi/ate_tests/supervisor_failure/Supervisor_Failure_test.go

```diff
@@ -0,0 +1,347 @@
+		WithSrcEndpoints(srcEndPoint).
+		WithDstEndpoints(dstEndPoint).
+		WithHeaders(ethHeader, ipv4Header)
+
+	ate.Traffic().Start(t, flow)
+	time.Sleep(15 * time.Second)
+	ate.Traffic().Stop(t)
+
+	flowPath := ate.Telemetry().Flow(flow.Name())
+	if got := flowPath.LossPct().Get(t); got > 0 {
+		t.Errorf("LossPct for flow %s got %g, want 0", flow.Name(), got)
+	} else {
+		t.Logf("Traffic flows fine from ATE-port1 to ATE-port2")
+	}
+}
+
+// testArgs holds the objects needed by a test case.
+type testArgs struct {
+	ctx     context.Context
+	clientA *gribi.Client
+	dut     *ondatra.DUTDevice
+	ate     *ondatra.ATEDevice
+	top     *ondatra.ATETopology
+}
+
+// routeInstall configures a IPv4 entry through clientA. Ensure that the entry via ClientA
+// is active through AFT Telemetry.
+func routeInstall(ctx context.Context, t *testing.T, args *testArgs) {
+	// Add an IPv4Entry for 203.0.113.0/24 pointing to ATE port-2 via gRIBI-A,
+	// ensure that the entry is active through AFT telemetry
+	t.Logf("Add an IPv4Entry for %s pointing to ATE port-2 via gRIBI-A", ateDstNetCIDR)
+	args.clientA.AddNH(t, nhIndex, atePort2.IPv4, instance, fluent.InstalledInRIB)
+	args.clientA.AddNHG(t, nhgIndex, map[uint64]uint64{nhIndex: 1}, instance, fluent.InstalledInRIB)
+	args.clientA.AddIPv4(t, ateDstNetCIDR, nhgIndex, instance, "", fluent.InstalledInRIB)
+
+	// Verify the entry for 203.0.113.0/24 is active through AFT Telemetry.
+	ipv4Path := args.dut.Telemetry().NetworkInstance(instance).Afts().Ipv4Entry(ateDstNetCIDR)
+	if got, want := ipv4Path.Prefix().Get(t), ateDstNetCIDR; got != want {
+		t.Errorf("ipv4-entry/state/prefix got %s, want %s", got, want)
+	}
+	// Verify that static route(203.0.113.0/24) to ATE port-2 is preferred by the traffic.`
+	srcEndPoint := args.top.Interfaces()[atePort1.Name]
+	dstEndPoint := args.top.Interfaces()[atePort2.Name]
+	testTraffic(t, args.ate, args.top, srcEndPoint, dstEndPoint)
+
+}
+
+// findComponentsByType finds supervisor (CONTROLLER_CARD) for switchover
+func findComponentsByType(t *testing.T, dut *ondatra.DUTDevice, cType telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT) []string {
+	components := dut.Telemetry().ComponentAny().Name().Get(t)
+	var s []string
+	for _, c := range components {
+		lookupType := dut.Telemetry().Component(c).Type().Lookup(t)
+		if !lookupType.IsPresent() {
+			t.Logf("Component %s type is not found", c)
+		} else {
+			componentType := lookupType.Val(t)
+			t.Logf("Component %s has type: %v", c, componentType)
+
+			switch v := componentType.(type) {
+			case telemetry.E_PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT:
+				if v == cType {
+					s = append(s, c)
+				}
+			default:
+				t.Fatalf("Expected component type to be a hardware component, got (%T, %v)", componentType, componentType)
+			}
+		}
+	}
+	return s
+}
+
+// findStandbyRP finds out ActiveRP and StandbyRP
+func findStandbyRP(t *testing.T, dut *ondatra.DUTDevice, supervisors []string) (string, string) {
```

Comment:
The RP/Supervisor naming convention is not OC compliant. Can we change these to `controller` which are either `primary` or `secondary`

Reply:
Which part you are referring.. 
 
Instead of standbyRP / activeRP, you want to use primary / secondary ? 

                if role == standbyController {
                        standbyRP = supervisor
                } else if role == activeController {
                        activeRP = supervisor
                } else {

Reply:
`findStandbyRP()` could be `findSecondaryController()`. 
Also,
standbyRP / activeRP --> primary/secondary.
activeController / standbyController --> primaryController / secondaryController.
supervisor --> controller

Reply:
These are internal function/variables names , following are OC compliant values. 

activeController  = telemetry.PlatformTypes_ComponentRedundantRole_PRIMARY
standbyController = telemetry.PlatformTypes_ComponentRedundantRole_SECONDARY

Reply:
Yup right. This is just a nit to make the code more vendor neutral.

The reasoning for changing the names to `primary/secondary` is that, `active/standby` and `primary/secondary` may not mean the same thing for some vendors. The types are defined here: https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-transport-line-protection.html#type-component-redundant-role

The same goes for `RP/supervisor` which are not defined and the only defined type is `controller_card`.

Reply:
ok. will change the variable names.

Reply:
Implemented.. New code push failed due to "Needs /gcbrun from a collaborator"..